data-structures.md

```diff
@@ -1945,6 +1945,33 @@ Slots are `setf`-able:
 ;; "Cookbook author"
 ~~~
 
+### symbol-macrolet
+
+If you are accessing several slots within a single function the
+special form `symbol-macrolet` can improve readibility, by creating
+symbol macros which expand into forms with structure accessors:
+
+~~~lisp
+(defstruct ship x-position y-position x-velocity y-velocity)
+
+(defun move-ship (ship)
+  (symbol-macrolet
+      ((x (ship-x-position ship))
+       (y (ship-y-position ship))
+       (xv (ship-x-velocity ship))
+       (yv (ship-y-velocity ship)))
+    (psetf x (+ x xv)
+           y (+ y yv))))
+~~~
+
+Here the math involved in the `move-ship` function is easier to
+read than if accessor functions were used.
+
+Though it is not mentioned in the standard, many modern implementations
+of Common Lisp permit the use of the CLOS macro `with-slots` with
```

Comment:
Can you precisely say which implementations?

Reply:
I wasn't sure how much to include about this, for three reasons. First, I wasn't sure how much time I wanted to spend installing all the CL implementations that'll run on my mac. Second, the `with-slots` thing fails (on SBCL, at least, and I imagine others) if you change the underlying representation of the struct to a list or a vector (a capability not discussed in this section). Finally, it's not technically part of the standard, even if people do use it.

I could rely on the kindness of strangers on discord to do a few tests for implementations I don't have straightforward access to.

Reply:
this is also good information that could appear in a subsection : )

\+1 to ask for feedback. Or just mention at least one implementation, for a start.